ui/src/containers/NodeVolumes.js

```diff
@@ -31,44 +43,134 @@ const VolumeTable = styled.div`
   margin-top: ${padding.small};
 `;
 
+const DeleteConfirmationModal = {
+  content: {
+    top: '50%',
+    left: '50%',
+    width: '540px',
+    height: '200px',
+    transform: 'translate(-50%, -50%)'
+  }
+};
+
+const NotificationButtonGroup = styled.div`
+  display: flex;
+  justify-content: flex-end;
+  margin-top: ${padding.larger};
+`;
+
+const DeleteButton = styled(Button)`
+  margin-left: ${padding.small};
+`;
+
+const IconButton = styled.button`
+  border-radius: 30px;
+  width: 30px;
+  height: 30px;
+  background-color: transparent;
+  border: none;
+  outline: none;
+  cursor: ${props => {
+    if (props.isEnableClick) {
+      return `pointer`;
+    } else {
+      return `not-allowed`;
+    }
+  }};
+
+  :hover {
+    background-color: ${props => {
+      if (props.isEnableClick) {
+        return grayLight;
+      }
+    }};
+  }
+`;
+
 const NodeVolumes = props => {
+  const { intl } = props;
+
   useRefreshEffect(refreshVolumesAction, stopRefreshVolumesAction);
   useRefreshEffect(
     refreshPersistentVolumesAction,
     stopRefreshPersistentVolumesAction
   );
+
   const volumes = useSelector(state => state.app.volumes);
+  const persistentVolumes = useSelector(state => state.app.volumes.pVList);
   const [sortBy, setSortBy] = useState('name');
   const [sortDirection, setSortDirection] = useState('ASC');
+  const [
+    isDeleteConfirmationModalOpen,
+    setisDeleteConfirmationModalOpen
+  ] = useState(false);
+  const [deleteVolumeName, setDeleteVolumeName] = useState('');
   const onSort = ({ sortBy, sortDirection }) => {
     setSortBy(sortBy);
     setSortDirection(sortDirection);
   };
 
+  const isVolumeDeletable = rowData => {
+    const volumeStatus = rowData.status;
+    const volumeName = rowData.name;
+
+    switch (volumeStatus) {
+      case STATUS_UNKNOWN:
+      case STATUS_PENDING:
+      case STATUS_TERMINATING:
+        return false;
+      case STATUS_FAILED:
+      case STATUS_AVAILABLE:
+        if (persistentVolumes?.length === 0) {
+          return true;
+        } else {
+          const persistentVolume = persistentVolumes.find(
+            pv => pv?.metadata?.name === volumeName
+          );
+          const persistentVolumeStatus = persistentVolume?.status?.phase;
+
+          switch (persistentVolumeStatus) {
+            case STATUS_FAILED:
+            case STATUS_AVAILABLE:
+            case STATUS_RELEASED:
+              return true;
+            case STATUS_PENDING:
```

Comment:
Hum, you'll want:

```javascript
            case STATUS_PENDING:
            case STATUS_BOUND:
              return false;
            default:
              console.error('Error! There is error from PV status.');
              return false;
```

Otherwise you'll log an error also for `STATUS_PENDING` and `STATUS_BOUND`: we don't want that.